lib/samson/hooks.rb

```diff
@@ -144,8 +144,10 @@ def fire(name, *args)
       end
 
       def render_views(name, view, *args)
-        hooks(name).each_with_object("".html_safe) do |partial, html|
-          html << view.render(partial, *args)
+        hooks(name).each_with_object("".html_safe) do |value, html|
+          partial, params_generator = value
+          new_args = params_generator ? [params_generator.call(*args)] : args
```

Comment:
The same template is shared by two different plugins which have different ways of computing the template parameters. e.g., one determines whether to show the form based on whether there's flowdock configuration and the other decides based on slack configuration.

Is there a different way to do it?